### PR TITLE
Created endpoint to allow admins to add coupon to user.

### PR DESCRIPTION
**Pull request by AzizPatel786**

### CODE REVIEW

File: server/src/service-layer/controllers/AdminController.ts [hunk 1]
```diff
@@ -329,4 +332,36 @@ export class AdminController extends Controller {
...
+        this.setStatus(404)
+        return
+      }
+      if (!user.stripe_id) {
+        this.setStatus(400)
+        return
+      }
+      console.log("Just before await")
+      // Add coupon to the user using Stripe ID
+      for (let i = 0; i < quantity; i++) {
```
Comment (bcho892) [inline server/src/service-layer/controllers/AdminController.ts:358]: Create an array of promises then use promise.all

File: server/src/business-layer/services/StripeService.ts [hunk 2]
```diff
@@ -450,4 +451,25 @@ export default class StripeService {
       console.error(`Failed to send an email to the user ${uid}`, error)
     }
   }
+
+  public async addCouponToUser(
+    stripeId: string,
+    amount: number
+  ): Promise<void> {
+    try {
+      const stripe = new Stripe(process.env.STRIPE_API_KEY)
```
Comment (bcho892) [inline server/src/business-layer/services/StripeService.ts:460]: Stripe is already declared as a global

**Review by bcho892 — CHANGES_REQUESTED** [review]
Just some small changes

File: server/src/service-layer/controllers/AdminController.ts [hunk 3]
```diff
@@ -329,4 +332,41 @@ export class AdminController extends Controller {
...
+        this.setStatus(404)
+        return
+      }
+      if (!user.stripe_id) {
+        this.setStatus(400)
+        return
+      }
+
+      // Add coupon to the user using Stripe ID
+      for (let i = 0; i < quantity; i++) {
```
Comment (bcho892) [inline server/src/service-layer/controllers/AdminController.ts:358]: This isn't needed anymore

File: server/src/service-layer/controllers/AdminController.ts [hunk 4]
```diff
@@ -329,4 +332,41 @@ export class AdminController extends Controller {
...
+        await stripeService.addCouponToUser(user.stripe_id, amount)
+      }
+
+      const couponPromises = Array.from({ length: quantity }, () =>
```
Comment (bcho892) [inline server/src/service-layer/controllers/AdminController.ts:362]: This isn't a promise, need to make it async
```suggestion
      const couponPromises = Array.from({ length: quantity }, async () =>
```

File: server/src/service-layer/controllers/AdminController.ts [hunk 5]
```diff
@@ -329,4 +332,41 @@ export class AdminController extends Controller {
...
+        stripeService.addCouponToUser(user.stripe_id, amount)
```
Comment (bcho892) [inline server/src/service-layer/controllers/AdminController.ts:363]:
```suggestion
       await stripeService.addCouponToUser(user.stripe_id, amount)
```

File: server/src/service-layer/controllers/AdminController.ts (hunk 3 above)
Comment (AzizPatel786) [reply server/src/service-layer/controllers/AdminController.ts:358]: Sorry forgot to remove it

**Review by bcho892 — APPROVED** [review]

**Pull request merged**